Fix ActivityLog recording when flagging a version because of an abuse report

diox opened the pull request:
When we're manually creating the `ActivityLog`, we need to pass a user for the `ActivityLog` to be saved. This code could be called from somewhere where there is no user in the thread, so explicitly fall back on the task user when that's happening.

Fixes mozilla/addons#2010

eviljeff reviewed (approved):
Can we have a line in one the existing tests that proves the activitylog is indeed created?

diox commented:
> Can we have a line in one the existing tests that proves the activitylog is indeed created?


`test_report()` in `TestCinderAddonHandledByReviewers` does it. It was passing because the test was doing a `set_user()` in `setUp()`, so I removed that, ensured test was failing without my fix, and also added another assertion verifying that `get_user()` returns `None` in that test now.
